crates/agglayer-config/src/storage.rs

```diff
@@ -0,0 +1,63 @@
+use std::path::PathBuf;
+
+use serde::Deserialize;
+use serde::Serialize;
+use serde_with::serde_as;
+
+/// Configuration for the storage.
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct StorageConfig {
+    /// Path to the metadata storage directory.
+    pub metadata_path: PathBuf,
+    /// Path to the pending storage directory.
+    pub pending_path: PathBuf,
+    /// Path to the state storage directory.
+    pub state_path: PathBuf,
+    /// Path to the epochs storage directory.
+    pub epochs_path: PathBuf,
+}
+
+impl Default for StorageConfig {
+    fn default() -> Self {
+        Self {
+            metadata_path: default_metadata_path(),
+            pending_path: default_pending_path(),
+            state_path: default_state_path(),
+            epochs_path: default_epochs_path(),
+        }
+    }
+}
+
+/// Returns the default path to the metadata storage directory.
+fn default_metadata_path() -> PathBuf {
+    match dirs::config_dir() {
```

Comment:
I was planning to use `config_dir` to store the database close to the config itself, so that we have one folder which contains everything related to the AggLayer.